src/js/modules/ListLink.js

```diff
@@ -11,7 +12,7 @@ class ListLink extends Component {
       <Link
         className={ `coa-ListLink ${(isBoxType && "coa-ListLink--boxprimary" )}` }
         key={id}
-        to={url}
+        to={getPathWithLangCode(url)}
```

Comment:
I like that you're trying to cover all app links to include language codes when present, but I think trying to manually build links with a helper this way is really fragile. We may forget to use this helper method/it requires a dev to KNOW to use a helper. Some links may come from Joplin's WYSIWYG editors or third party content.  Maybe we can build a component wrapper for router Link components that do the language modification. And then devs only have to remember to use Link components for Links and this change is made in 1 place rather than many.

Reply:
Yup yup. Agreed on fragile and easy to forget. 

For Joplin/3rd party content, I'm not sure that we should intervene and change links in that content. That seems a little unwieldy in a different way. But you're right, it should be a consideration. That type of parsing may also be a responsible we could give to the server. That would do the job more holistically for a world where the authored content goes to many clients, not just Janis.

As far as your suggestion around building a component wrapper for the Link... Are you saying, we write a new component called `<Link>` that itself imports the react-router-dom version of `<Link>` and we just move this `getPathWithLangCode` function into that component? That seems like a solid plan to me. Then the onus on the developer is to remember to import Link from the module we've created instead of from the standard library. Should we call it Link? Or LanguageLink? what do you think?